api/controllers/subscription.js

```diff
@@ -0,0 +1,105 @@
+const Subscription = require('../models/subscriptions');
+
+module.exports = {
+  createSubscription,
+  getSubscription,
+  updateSubscription,
+  deleteSubscription,
+};
+
+function createSubscription(req, res) {
+  const subscriptionId = req.swagger.params.subscriptionId.value;
+  const newSubscription = req.body;
+  newSubscription.subscriptionId = subscriptionId;
+  const subscription = new Subscription(newSubscription);
+  subscription.save(function(err, subscription) {
+    if (err) {
+      console.error('error', err);
+      return res.status(409).json({
+        message: 'Error saving subscription',
+        error: err.message,
+      });
+    }
+    return res.status(200).json(subscription.toJSON());
+  });
+}
+
+function getSubscription(req, res) {
+  const subscriptionId = req.swagger.params.subscriptionId.value;
+
+  Subscription.findOne({ subscriptionId }, function(err, subscription) {
+    if (err) {
+      return res.status(500).json({
+        message: 'Error getting Subscription. ',
+        error: err.message,
+      });
+    }
+    if (!subscription) {
+      return res.status(404).json({
+        message:
+          'Subscription does not exist. Subscription Id: ' + subscriptionId,
+      });
+    }
+    return res.status(200).json(subscription.toJSON());
+  });
+}
+
+function updateSubscription(req, res) {
+  const subscriptionId = req.swagger.params.subscriptionId.value;
+
+  Subscription.findOne({ subscriptionId }, function(err, subscription) {
+    if (err) {
+      return res.status(500).json({
+        message: 'Error updating subscription. ',
+        error: err.message,
+      });
+    }
+    if (!subscription) {
+      return res.status(404).json({
+        message:
+          'Subscription does not exist. Subscription Id: ' + subscriptionId,
+      });
+    }
+    for (let key in req.body) {
+      subscription[key] = req.body[key];
+    }
+    subscription.save(function(err, subscription) {
+      if (err) {
+        return res.status(500).json({
+          message: 'Error saving subscription. ',
+          error: err.message,
+        });
+      }
+      if (!subscription) {
+        return res.status(404).json({
+          message:
+            'Unable to find subscription. subscription id: ' + subscriptionId,
+        });
+      }
+      return res.status(200).json(subscription.toJSON());
+    });
+  });
+}
+
+function deleteSubscription(req, res) {
+  const subscriptionId = req.swagger.params.subscriptionId.value;
+
+  Subscription.findOneAndDelete({ subscriptionId }, function(
+    err,
+    subscription
+  ) {
+    if (err) {
+      return res.status(500).json({
+        message: 'Error deleting subscription. ',
+        error: err.message,
+      });
+    }
+    if (!subscription) {
+      return res.status(404).json({
+        message:
+          'Subscription does not exist. Subscription Id: ' + subscriptionId,
+      });
+    }
+    return res.status(200).json(subscription.toJSON());
+  });
+}
```

Comment:
@tomivm Subscription related operations can be done exclusively by an admin user.

Reply:
@martinbedouret Subscription operations are only allowed for admin users. Check the values of x-security-scopes for Subscription operations on the swagger.yaml : https://github.com/cboard-org/cboard-api/blob/caf7646b11f7816eda59b4dba82e214e31f61ec9/api/swagger/swagger.yaml#L890

Also there are some tests to check this. please see: https://github.com/cboard-org/cboard-api/blob/caf7646b11f7816eda59b4dba82e214e31f61ec9/test/controllers/subscription.js#L55
Its are commented because before merging #225 Use admins endpoints with no token or user token cause a crash of entire API. I will uncomment its on nexts commit's.